CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+- `load`: Function which when called will load and initialize the provider
+  script. It should accept props and use the `providerKey` as the `app_id` or
+  `api_key`. For consistency, it should call a `loadScript` function.
+- `open`: Function which when called will open the provider chat.
+- `close`: Function which when called will close the provider chat.
+
+Import the new file in `src/providers/index.js` and add it to `Providers`.
+
+The name of this file will be the `providerKey` used in the
+`LiveChatLoaderProvider`.
+
+### 2. Create component
+
+Create a new component in `src/Components/ProviderName/index.js` which
+replicates the chat widget, using the following as a template:
+
+<details>
+<summary>Component Template</summary>
+
+```jsx
+import React from 'react'
+
+import { useChat } from '../../'
+import STATES from '../../utils/states'
+
+const styles = {
+  // Add widget styles here
+  button: {
+    // Add button styles here
+  }
+}
+
+const Provider = ({ color }) => {
+  const [state, loadChat] = useChat({ loadWhenIdle: true })
+
+  if (state === STATES.COMPLETE) return null
+
+  return (
+    <div>
+      <button
+        onClick={() => loadChat({ open: true })}
+        onMouseEnter={() => loadChat({ open: false })}
+        style={{
+          ...styles.button,
+          backgroundColor: color
+        }}
+      >
+        Button
+      </button>
+    </div>
+  )
+}
+
+Provider.defaultProps = {
+  color: '#976ad4'
+}
+
+export default Provider
+```
+
+</details>
+
+Do not worry about loading animations as the widget
+will be shown instantly on page load. Increase the `z-index` by `1` so the fake
+widget sits immediately above the chat widget that is being replaced.
+
+Export the component from `src/index.js`
+
+### 3. Update README
+
+Add your new provider to this README under [Supported Providers](#supported-providers).
+
+### 4. Add an example page
+
+Add a new page to `website/pages/` which showcases the provider. If you don't want to include your `providerKey` leave this blank and the maintainers will set one up.
+
+The new provider page can be tested locally by creating a distribution version of the package and referencing this from the `website`.
+
+Unfortunately if you try to include the package locally from source you'll most likely run into a [Duplicate React](https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react) error.
+
+To create the distribution version and reference it, do the following:
+
+- In the root of the project, run `npm run build` to build the package into `dist`
+- Update `website/package.json` to reference the `dist` build: `"react-live-chat-loader": "../"`
+- In the `website` directory run `npm install`
+- In the `website` directory run the server with `npm run dev`
+- Add a new page to `website/pages/` which includes the new component
+- Add a link to the provider in `website/pages/index.js`
+- Add a link to the provider in `website/components/exampleLinks.js`
```

Comment:
While we're here we can update this section:

```suggestion
### 4. Add an example page

Add a new page to `website/pages/` which showcases the provider. If you don't want to include your `providerKey` leave this blank and the maintainers will set one up.

The new provider page can be tested locally by creating a distribution version of the package and referencing this from the `website`.

To create the distribution version and reference it, do the following:

- In the root of the project, run `npm run build`
- In the `website` directory run `npm install`
- In the `website` directory run the server with `npm run dev`
- Add a new page to `website/pages/` which includes the new component
- Add a link to the provider in `website/pages/index.js`
- Add a link to the provider in `website/components/exampleLinks.js`
```

If you run into a  [Duplicate React](https://reactjs.org/warnings/invalid-hook-call-warning.html#duplicate-react) error you can run `npm link website/node_modules/react` from the library directory.